### PR TITLE
[KED-2923] Add deprecation warning to --parallel flag

### PR DESCRIPTION
**Pull request by AntonyMilneQB**
## Description
In kedro 0.18, `--parallel` will be removed in favour of `--runner=ParallelRunner`.

Here's the warning you get with `--parallel` or `-p` now:
image

I checked the whole CLI tree in case there were any other glaring things we could easily improve for 0.18 and didn't find anything else, so I think we're good 🙂 

Actual removal and changes to docs in #1030.

## Checklist

- [x] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [x] Added tests to cover my changes

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: kedro/framework/cli/project.py [hunk 1]
```diff
@@ -46,9 +46,11 @@
 RUNNER_ARG_HELP = """Specify a runner that you want to run the pipeline with.
 Available runners: `SequentialRunner`, `ParallelRunner` and `ThreadRunner`.
 This option cannot be used together with --parallel."""
-PARALLEL_ARG_HELP = """Run the pipeline using the `ParallelRunner`.
+PARALLEL_ARG_HELP = """(DEPRECATED) Run the pipeline using the `ParallelRunner`.
 If not specified, use the `SequentialRunner`. This flag cannot be used together
-with --runner."""
+with --runner. In Kedro 0.18.0, `-p` will be an alias for `--pipeline` and the 
+`--parallel` flag will no longer exist. Instead, the parallel runner should be used by 
+specifying `--runner=ParallelRunner` (or `-r`)."""
```
Comment (AntonyMilneQB) [inline kedro/framework/cli/project.py:53]:
```suggestion
specifying `--runner=ParallelRunner` (or `-r ParallelRunner`)."""
```

File: kedro/framework/cli/project.py [hunk 2]
```diff
@@ -372,6 +374,13 @@ def run(
         )
     runner = runner or "SequentialRunner"
     if parallel:
+        deprecation_message = (
+            "DeprecationWarning: The behaviour of --parallel and -p flags will change. "
+            "In Kedro 0.18.0, `-p` will be an alias for `--pipeline` and the "
+            "`--parallel` flag will no longer exist. Instead, the parallel runner "
+            "should be used by specifying `--runner=ParallelRunner` (or `-r`)."
```
Comment (AntonyMilneQB) [inline kedro/framework/cli/project.py:381]:
```suggestion
            "should be used by specifying `--runner=ParallelRunner` (or "
            "`-r ParallelRunner`)."
```

File: kedro/framework/cli/project.py [hunk 3]
```diff
@@ -372,6 +374,14 @@ def run(
         )
     runner = runner or "SequentialRunner"
     if parallel:
+        deprecation_message = (
+            "DeprecationWarning: The behaviour of --parallel and -p flags will change. "
```
Comment (datajoely) [inline kedro/framework/cli/project.py:378]: Do we want to mention the thinking that we want to `-p` to fall under the `--pipeline` category?

Comment (AntonyMilneQB) [reply kedro/framework/cli/project.py:378]: Already there in the next line 😀

**Review by MerelTheisenQB — APPROVED** [review]
LGTM! 👍

**Review by lorenabalan — APPROVED** [review]

**Pull request merged**